### PR TITLE
0.5 release

### PR DESCRIPTION
**Pull request by riveraj**
See [CHANGELOG](https://github.com/ubccr/ralph/blob/0.5/CHANGELOG.md) for summary of changes.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,33 +1,31 @@
...
 	@ echo "Targets:"
-	@ echo "  all               Clean, build, verify, test and sniff the code."
-	@ echo "                    This target should succeed prior to opening a"
-	@ echo "                    pull request."
-	@ echo "  build             Builds the project."
-	@ echo "  clean             Clean all artifacts managed by make."
-	@ echo "  lint              Run the PHP linter on all .php files."
-	@ echo "  phpcs             Run PHP_CodeSniffer."
-	@ echo "  test              Run unit tests."
+	@ echo "  analyze Run the PHPStan static analysis tool on the codebase."
```
Comment (smgallo) [inline Makefile:4]: Nice set of make targets.

File: composer.json [hunk 2]
```diff
@@ -2,12 +2,12 @@
...
     "description": "A simple SPARQL client built on top of Guzzle.",
     "type": "library",
-    "keywords": ["sparql","rdf"],
+    "keywords": ["rdf", "sparql"],
     "homepage": "https://github.com/ubccr/ralph",
     "license": "LGPL-3.0-only",
     "authors": [
         {
-            "name": "Center for Computational Research, the University at Buffalo",
+            "name": "Center for Computational Research, SUNY the University at Buffalo",
```
Comment (smgallo) [inline composer.json]: The official name is "Center for Computational Research, University at Buffalo. The State University of New York"

File: src/Client.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+     *
+     * @return self This instance to allow for method chaining.
+     *
+     * @throws InvalidUriException If the namespace is not a valid RFC 3986 URI.
+     */
+
+    public function withPrefix(string $namespace, string $uri): self
+    {
+        $this->validate($uri);
+        $this->prefixes[] = [$namespace, $uri];
```
Comment (smgallo) [inline src/Client.php]: Should namespace also be validated? Can it be empty?

File: src/ResultSet.php [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+ */
+
+class ResultSet implements IteratorAggregate, JsonSerializable
+{
+    /** @var array Results returned from a query. */
+    private $results;
+
+    public function __construct(string $results)
+    {
+        $this->results = json_decode($results, true);
```
Comment (smgallo) [inline src/ResultSet.php]: Do you want to check that `json_decode` didn't return `null` or is that handled elsewhere?

File: src/ResultSet.php [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+    /**
+     * Get the variables included in the results.
+     *
+     * @return array The variables as an array.
+     */
+
+    public function getVariables(): array
+    {
+        return $this->results['head']['vars'];
```
Comment (smgallo) [inline src/ResultSet.php]: What if these elements aren't set? Should you return `false` or `null` or throw an exception in that case?

**Review by smgallo — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (riveraj) [reply composer.json]: I get the distinct feeling there was a serious meeting where a committee debated the formatting of that name.

"Should that be a comma here, or a period?"

**Review by smgallo — APPROVED** [review]

**Pull request merged**